pkg/util/merge/merge_podtemplate_spec.go

```diff
@@ -0,0 +1,237 @@
+package merge
+
+import (
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/contains"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func PodTemplateSpecs(original, override corev1.PodTemplateSpec) corev1.PodTemplateSpec {
+	merged := original
+
+	merged.Labels = StringToStringMap(original.Labels, override.Labels)
+	merged.Spec.Volumes = Volumes(original.Spec.Volumes, override.Spec.Volumes)
+	merged.Spec.Containers = Containers(original.Spec.Containers, override.Spec.Containers)
+	merged.Spec.InitContainers = Containers(original.Spec.InitContainers, override.Spec.InitContainers)
+
+	if override.Spec.EphemeralContainers != nil {
+		merged.Spec.EphemeralContainers = EphemeralContainers(original.Spec.EphemeralContainers, override.Spec.EphemeralContainers)
```

Comment:
[nit] I just now think that it would probably be cleaner to have the nil check inside the `EphemeralContainers` function. But I should have probably noted this before 😅 as I think we might have a few other similar patterns and it is not worth refactoring all of them now

Reply:
yes maybe you're right! We can maybe do one last pass after the merging logic is all there to tidy up things like this!